### PR TITLE
Minor QOL improvements I desired

### PR DESCRIPTION
**Pull request by b0dee**
Addressed a number of improvements I desired: 

- [x] Launch on start
- [x] Toggle scroll bar
- [x] Customise highlight group

### CODE REVIEW

**Comment by noscript** [comment]
Thanks for the PR. Could you please also add new variables and command to the documentation?

**Comment by b0dee** [comment]
Documentation updated, thanks for accepting it and great work on the plugin!

**Comment by b0dee** [comment]
Submitted another as noticed the scroll bar wouldn't reach end of window until topline == last line in file
It will now put it to bottom of screen the moment the last line of file is on the window (if it wasn't going to already, not sure if edge case but happened a bit in files I was testing on)

File: autoload/elevator.vim [hunk 1]
```diff
@@ -7,6 +7,16 @@ endif
 if !exists('g:elevator#width')
   g:elevator#width = 1
 endif
+if !exists('g:elevator#show_on_enter')
+  g:elevator#width = v:false
```
Comment (noscript) [inline autoload/elevator.vim]: This condition unnecessary. Besides it breaks the compilation as `g:elevator#width` is boolean.

File: autoload/elevator.vim [hunk 2]
```diff
@@ -7,6 +7,16 @@ endif
...
+endif
+if !exists('g:elevator#highlight')
+  g:elevator#highlight = ''
+endif
+
+if !exists('g:elevator#hidden')
+  g:elevator#hidden = v:false
```
Comment (noscript) [inline autoload/elevator.vim]: Please use `false`.

File: autoload/elevator.vim [hunk 3]
```diff
@@ -33,8 +43,18 @@ def S__calculate_scale(winid__a: number): float
   return 1.0 * winheight / (line('$', winid__a) + winheight)
 enddef
 
+export def Toggle()
+  if g:elevator#hidden
+    g:elevator#hidden = v:false
```
Comment (noscript) [inline autoload/elevator.vim]: `true`, `false`. Optionally `g:elevator#hidden = !g:elevator#hidden`

File: doc/elevator.txt [hunk 4]
```diff
@@ -40,3 +41,38 @@ Example:
 >
     g:elevator#width = 2
 <
+
+*g:elevator#highlight*                Set the highlight group for the scroll bar
```
Comment (noscript) [inline doc/elevator.txt]: Please put `.` at the end of sentences, for consistency with the rest of documentation.

File: plugin/elevator.vim [hunk 5]
```diff
@@ -8,5 +8,12 @@ g:elevator_loaded = true
 import autoload 'elevator.vim'
 
 augroup Elevator
-  autocmd WinScrolled * elevator#Show(expand('<amatch>')->str2nr())
+	autocmd!
+	if exists('g:elevator#show_on_enter') && g:elevator#show_on_enter == v:true
```
Comment (noscript) [inline plugin/elevator.vim]: `if exists('g:elevator#show_on_enter') && g:elevator#show_on_enter`

File: autoload/elevator.vim [hunk 6]
```diff
@@ -96,7 +119,12 @@ export def S__set_geometry()
   endif
 
   var popup_height = S__clamp((wininfo.height * scale)->round()->float2nr(), 1, wininfo.height)
-  var popup_offset = S__clamp(((wininfo.topline - 1) * scale)->round()->float2nr(), 0, wininfo.height - popup_height)
+  var popup_offset = 0
+  if wininfo.botline == line('$', s_state.scrolled_winid)
+    popup_offset = wininfo.height - popup_height
+  else
+    popup_offset = S__clamp(((wininfo.topline - 1) * scale)->round()->float2nr(), 0, wininfo.height - popup_height)
+  endif
```
Comment (noscript) [inline autoload/elevator.vim]: This change makes the scrollbar not move if buffer is scrolled below the last line, and then suddenly jump up when it's scrolled above the last line.

**Review by noscript — CHANGES_REQUESTED** [review]

**Comment by noscript** [comment]
> Submitted another as noticed the scroll bar wouldn't reach end of window until topline == last line in file It will now put it to bottom of screen the moment the last line of file is on the window (if it wasn't going to already, not sure if edge case but happened a bit in files I was testing on)

This is intended and follows the behavior of GVim scrollbars (`guioptions+=r`), because Vim allows to scroll the buffer past the last line.

File: autoload/elevator.vim (hunk 1 above)
Comment (b0dee) [reply autoload/elevator.vim]: Sorry - bug, should be g:elevator#show_on_enter = false
Fixed in next PR

File: autoload/elevator.vim (hunk 2 above)
Comment (b0dee) [reply autoload/elevator.vim]: Fixing in next PR

File: autoload/elevator.vim (hunk 3 above)
Comment (b0dee) [reply autoload/elevator.vim]: Neatened and used toggling boolean (!)
Fixes in next PR

File: doc/elevator.txt (hunk 4 above)
Comment (b0dee) [reply doc/elevator.txt]: Resolved in next PR

File: plugin/elevator.vim (hunk 5 above)
Comment (b0dee) [reply plugin/elevator.vim]: Fixed in next PR

**Comment by b0dee** [comment]
Ahh - learn something new everyday. All changes completed submitting PR now

**Comment by noscript** [comment]
I rebased and merged the remaining changes manually.

**Pull request closed without merging**